src/Chainweb/Pact/SPV.hs

```diff
@@ -224,3 +234,64 @@ getTxIdx bdb pdb bh th = do
 
     sindex :: Monad m => (a -> Bool) -> S.Stream (S.Of a) m () -> m (Maybe Natural)
     sindex p s = S.zip (S.each [0..]) s & sfind (p . snd) & fmap (fmap fst)
+
+
+-- | Encode a "successful" CommandResult into a Pact object.
+mkSPVResult
+    :: CommandResult Hash
+       -- ^ Full CR
+    -> PactValue
+       -- ^ Success result
+    -> Object Name
+mkSPVResult CommandResult{..} j =
+    Object (ObjectMap $ M.fromList $
+            [ ("result", fromPactValue j)
+            , ("req-key", tStr $ asString $ unRequestKey _crReqKey)
+            , ("txid", tStr $ maybe "" asString _crTxId)
+            , ("gas", toTerm $ (fromIntegral _crGas :: Integer))
+            , ("meta", maybe empty metaField _crMetaData)
+            , ("logs", tStr $ asString $ _crLogs)
+            , ("continuation", maybe empty contField _crContinuation)
+            , ("events", toTList TyAny def $ map eventField _crEvents)
+            ])
+    TyAny Nothing def
+  where
+    metaField v = case fromJSON v of
```

Comment:
@larskuhtz so this field is a little annoying -- `_crMetaData` is `Maybe Value`, as it's supposed to be an entirely platform-dependent free field in the Pact API. Here, `fromJSON` is converting to a `PactValue`, and note that if it fails, it simply adds an empty object.